tests/test_distributions.py

```diff
@@ -179,16 +179,20 @@ def test_check_distribution_compatibility() -> None:
         ),
     )
 
-    # test dynamic value range (except CategoricalDistribution)
+    # Test dynamic value range (except CategoricalDistribution).
     distributions.check_distribution_compatibility(
         EXAMPLE_DISTRIBUTIONS["i"], distributions.IntDistribution(low=-3, high=2)
     )
     distributions.check_distribution_compatibility(
         EXAMPLE_DISTRIBUTIONS["il"], distributions.IntDistribution(low=1, high=13, log=True)
     )
-    distributions.check_distribution_compatibility(
-        EXAMPLE_DISTRIBUTIONS["id"], distributions.IntDistribution(low=-3, high=2, step=2)
-    )
+    with warnings.catch_warnings():
+        # When `step` is 2, UserWarning will be raised since the range is not divisible by 2.
+        # The range will be replaced with [-3, 1].
+        warnings.simplefilter("ignore", category=UserWarning)
+        distributions.check_distribution_compatibility(
+            EXAMPLE_DISTRIBUTIONS["id"], distributions.IntDistribution(low=-3, high=2, step=2)
```

Comment:
How about using `low=-3` and `high=1` to suppress `UserWarning`?

Reply:
Thanks! Definitely, it looks simpler.